optimizer/src/main/java/com/netease/arctic/optimizer/operator/executor/BaseExecutor.java

```diff
@@ -50,4 +75,15 @@ protected long getMaxTransactionId(List<DataFile> dataFiles) {
 
     return 0;
   }
+
+  protected void checkIfTimeout(Closeable writer) throws Exception {
+    long maxExecuteTime = task.getMaxExecuteTime() != null ?
+        task.getMaxExecuteTime() : TableProperties.OPTIMIZE_EXECUTE_TIMEOUT_DEFAULT;
+    if (System.currentTimeMillis() - startTime > maxExecuteTime * factor) {
+      writer.close();
+      LOG.info("table {} execute task {} timeout, max execute time is {}",
+          table.id(), task.getTaskId(), task.getMaxExecuteTime());
+      throw new TimeoutException("optimizer execute timeout");
```

Comment:
`TimeoutException`'s message should contain more information like 'actual execute time', 'max execute time', and 'factor'.

Reply:
ok